### PR TITLE
Back End

### PR DESCRIPTION
**Pull request by AriLFrankel**
Mongo / Mongoose persistence layer
Schema
RESTful API
Controllers

### CODE REVIEW

File: server/config/db.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose');
+
+mongoose.Promise = global.Promise;
```
Comment (rossjs) [inline server/config/db.js:3]: necessary?

File: server/router.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const router = require('express').Router();
+const { user, event } = require('./controllers');
+
+router.get('/user', user.getAllUsers);
```
Comment (rossjs) [inline server/router.js:4]: Does the router automatically send back a 200 if no error is caught?

**Review by rossjs — COMMENTED** [review]
@AriLFrankel Just have two questions in there regarding mongoose.Promise and how Express Router responds back to the client.

File: server/config/db.js (hunk 1 above)
Comment (KellyJohnBraun) [reply server/config/db.js:3]: Mongoose native library may be deprecated (as of February)

according to the docs, you need another line 
 ``` javascript
// Use native promises
    mongoose.Promise = global.Promise;
    assert.equal(query.exec().constructor, global.Promise);
``` 


http://mongoosejs.com/docs/promises.html

File: server/router.js (hunk 2 above)
Comment (rossjs) [reply server/router.js:4]: The answer to this was "yes"

Also, hi Kelly!

**Pull request merged**